.evergreen/compile_with_libmongocrypt.sh

```diff
@@ -0,0 +1,252 @@
+# to this script will be forwarded on as flags passed to cmake.
+#
+# This script should be run from the root of the repository.  This script will run the build from
+# the default build directory './build'.  The following environment variables will change the
+# behavior of this script:
+# - build_type: must be set to "Release" or "Debug"
+
+set -o errexit
+set -o pipefail
+
+: "${branch_name:?}"
+: "${build_type:?}"
+: "${distro_id:?}" # Required by find-cmake-latest.sh.
+
+: "${COMPILE_MACRO_GUARD_TESTS:-}"
+: "${ENABLE_CODE_COVERAGE:-}"
+: "${ENABLE_TESTS:-}"
+: "${generator:-}"
+: "${platform:-}"
+: "${REQUIRED_CXX_STANDARD:-}"
+: "${RUN_DISTCHECK:-}"
+: "${USE_POLYFILL_BOOST:-}"
+: "${USE_SANITIZER_ASAN:-}"
+: "${USE_SANITIZER_UBSAN:-}"
+: "${USE_STATIC_LIBS:-}"
+
+mongoc_prefix="$(pwd)/../mongoc"
+echo "mongoc_prefix=${mongoc_prefix:?}"
+
+if [[ "${OSTYPE:?}" =~ cygwin ]]; then
+  mongoc_prefix=$(cygpath -m "${mongoc_prefix:?}")
+fi
+
+# shellcheck source=/dev/null
+. "${mongoc_prefix:?}/.evergreen/scripts/find-cmake-latest.sh"
+export cmake_binary
+cmake_binary="$(find_cmake_latest)"
+command -v "$cmake_binary"
+
+if [ ! -d ../drivers-evergreen-tools ]; then
+  git clone --depth 1 https://github.com/mongodb-labs/drivers-evergreen-tools.git ../drivers-evergreen-tools
+fi
+# shellcheck source=/dev/null
+. ../drivers-evergreen-tools/.evergreen/find-python3.sh
+# shellcheck source=/dev/null
+. ../drivers-evergreen-tools/.evergreen/venv-utils.sh
+
+venvcreate "$(find_python3)" venv
+python -m pip install GitPython
+
+if [[ "${build_type:?}" != "Debug" && "${build_type:?}" != "Release" ]]; then
+  echo "$0: expected build_type environment variable to be set to 'Debug' or 'Release'" >&2
+  exit 1
+fi
+
+if [[ "${OSTYPE}" == darwin* ]]; then
+  # MacOS does not have nproc.
+  nproc() {
+    sysctl -n hw.logicalcpu
+  }
+fi
+CMAKE_BUILD_PARALLEL_LEVEL="$(nproc)"
+export CMAKE_BUILD_PARALLEL_LEVEL
+
+# Use ccache if available.
+if [[ -f "${mongoc_prefix:?}/.evergreen/scripts/find-ccache.sh" ]]; then
+  # shellcheck source=/dev/null
+  . "${mongoc_prefix:?}/.evergreen/scripts/find-ccache.sh"
+  find_ccache_and_export_vars "$(pwd)" || true
+fi
+
+
+build_targets=()
+cmake_build_opts=()
+case "${OSTYPE:?}" in
+cygwin)
+  cmake_build_opts+=("/verbosity:minimal")
+  build_targets+=(--target ALL_BUILD --target examples/examples)
+  ;;
+
+darwin* | linux*)
+  build_targets+=(--target all --target examples)
+  ;;
+
+*)
+  echo "unrecognized operating system ${OSTYPE:?}" 1>&2
+  exit 1
+  ;;
+esac
+
+# Create a VERSION_CURRENT file in the build directory to include in the dist tarball.
+python ./etc/calc_release_version.py >./build/VERSION_CURRENT
+cd build
+
+cmake_flags=(
+  "-DCMAKE_BUILD_TYPE=${build_type:?}"
+  "-DCMAKE_PREFIX_PATH=${mongoc_prefix:?}"
+  -DBUILD_TESTING=ON
+  -DMONGOCXX_ENABLE_SLOW_TESTS=ON
+  -DCMAKE_INSTALL_PREFIX=install
+  -DENABLE_UNINSTALL=ON
+  -DENABLE_CLIENT_SIDE_ENCRYPTION=ON
```

Comment:
`ENABLE_CLIENT_SIDE_ENCRYPTION` only applies to the C driver build, not C++. Try applying this commit: https://github.com/mongodb/mongo-cxx-driver/commit/089dc3f0ade18a1cc7c82ef47dbab587f8f53a7e
 to remove this file and the `compile_with_libmongocrypt` Evergreen task.